src/disco/net/xdp/fd_xdp_tile.c

```diff
@@ -340,6 +356,51 @@ net_is_fatal_xdp_error( int err ) {
          err==EPERM;
 }
 
+/* Load the netdev table to ctx->netdev_buf */
+
+static void
+net_load_netdev_tbl( fd_net_ctx_t * ctx ) {
+  /* Copy from double buffer */
+  if( FD_UNLIKELY( fd_dbl_buf_read( ctx->netdev_dbl_handle, ctx->netdev_buf_sz, ctx->netdev_buf, NULL ) ) == 0)  FD_LOG_ERR(("netdev table load failed"));
+  
+  /* Recalculate the join handler to netdeb buf */
+  if( FD_UNLIKELY( !fd_netdev_tbl_join( &ctx->netdev_tbl_handle, ctx->netdev_buf ) ) ) FD_LOG_ERR(("netdev table join failed"));
+}
+
+/* Query the netdev table. Return a fd_netdev_t pointer to the net device of the 
+interface specified by if_idx. Null if the if_idx is invalid */
+
+static fd_netdev_t *
+net_query_netdev_tbl( fd_net_ctx_t *ctx, 
+                      uint if_idx ) {  
+  if( FD_UNLIKELY( !fd_netdev_tbl_join( &ctx->netdev_tbl_handle, ctx->netdev_buf ) ) ) FD_LOG_ERR(("netdev table join failed"));
+  
+  /* dev_tbl is one-indexed */
+  if( if_idx>ctx->netdev_tbl_handle.hdr->dev_cnt ) {
+    FD_LOG_NOTICE(( "interface idx invalid (%u). dev_cnt (%u)", if_idx, ctx->netdev_tbl_handle.hdr->dev_cnt ));
+    return NULL;
+  } 
+
+  return &ctx->netdev_tbl_handle.dev_tbl[if_idx];
+}
+
+/* Iterates the netdev table and returns 1 if a GRE interface exists, 0 otherwise.
+   Only called in privileged_init and during_housekeeping */
+
+static int
+net_check_GRE_interface_exists( fd_net_ctx_t * ctx ) {
+  if( FD_UNLIKELY( !fd_netdev_tbl_join( &ctx->netdev_tbl_handle, ctx->netdev_buf ) ) ) FD_LOG_ERR(("netdev table join failed"));
```

Comment:
Can you do this join in load table instead? It's nice to always have a valid join around in case we need it